extension/src/paymentHandler/payment-handler.js

```diff
@@ -39,9 +41,31 @@ function _getPaymentHandlers (paymentObject) {
     handlers.push(getPaymentMethodsHandler)
   if (paymentObject.custom.fields.makePaymentRequest && !paymentObject.custom.fields.makePaymentResponse)
     handlers.push(makePaymentHandler)
+  if (paymentObject.custom.fields.makePaymentResponse
+    && paymentObject.custom.fields.submitAdditionalPaymentDetailsRequest
+    && isSubmitAdditionalPaymentDetailsRequestNew(paymentObject))
+    handlers.push(submitPaymentDetailsHandler)
   return handlers
 }
 
+function isSubmitAdditionalPaymentDetailsRequestNew (paymentObject) {
+  const interfaceInteraction = paymentObject.interfaceInteractions
```

Comment:
In this method, the reliability could be improved, return first doing another thing like if no interface interaction return true, and not put it after 20 lines.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/271/commits/d8f9e9d5f48509678153ad8ba2749d861ee46850#diff-40aa86c61762b743c388abe89395480cR39-R43